metrics/influxdb.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""InfluxDB client that pushes time series points from JSON"""
+
+import httplib
+import json
+import time
+import urllib
+
+class Point(object):
+    """Represents an InfluxDB time series point and handles serialization."""
+    def __init__(self, measurement, tags, fields, time_stamp):
+        """Creates a new Point"""
+        self.measurement = measurement
+        self.tags = tags
+        self.fields = fields
+        self.time = time_stamp
+
+    @staticmethod
+    def from_dict(dct):
+        """Deserialization function to be used by json.load()."""
+        if ('measurement' in dct and
+                'tags' in dct and
+                'fields' in dct):
+            return Point(
+                dct['measurement'],
+                dct['tags'],
+                dct['fields'],
+                dct.get('time'),
+            )
+        return dct
+
+
+    def __repr__(self):
```

Comment:
It seems like neither \_\_repr__ nor \_\_str__ are really appropriate since the function is really serializing the object into influxdb line protocol syntax as opposed to displaying a python representation of the object or a human readable string representation.  I changed the name of the function to `serialize` to distinguish this.